cmd/object-api-utils.go

```diff
@@ -64,12 +69,13 @@ func IsValidBucketName(bucket string) bool {
 	if len(bucket) < 3 || len(bucket) > 63 {
 		return false
 	}
-	if bucket[0] == '.' || bucket[len(bucket)-1] == '.' {
```

Comment:
For all valid bucket names this check would be false and we would be evaluating regex matches in all these cases. It isn't worth optimising for invalid bucket names.